File: fixtures/scale/java/small/src/textprinter/util/TextUtil.java
package textprinter.util;

public class TextUtil {
    public static String padTo(String line, int width) {
        if (line.length() >= width) {
            return line;
        }
        return line + " ".repeat(width - line.length());
    }

    public static String repeatChar(char c, int count) {
        StringBuilder sb = new StringBuilder(count);
        for (int i = 0; i < count; i++) {
            sb.append(c);
        }
        return sb.toString();
    }

    public static int visibleLength(String line) {
        return line.replaceAll("\\[[0-9;]*m", "").length();
    }
}

File: fixtures/scale/dart/small/lib/util/text_util.dart
String padTo(String line, int width) =>
    line.length >= width ? line : line + (' ' * (width - line.length));

String repeatChar(String c, int count) => c * count;

int visibleLength(String line) =>
    line.replaceAll(RegExp(r'ESC\[[0-9;]*m'), '').length;

String truncate(String line, int width) =>
    line.length <= width ? line : line.substring(0, width);
